packages/contracts-bedrock/test/invariants/symbolic/OptimismSuperchainERC20.t.sol

```diff
@@ -0,0 +1,221 @@
+
+    function deal(address, address, uint256) external;
+}
+
+contract HalmosTest is SymTest, Test { }
+
+contract OptimismSuperchainERC20_SymTest is HalmosTest {
+    uint256 internal constant CURRENT_CHAIN_ID = 1;
+    uint256 internal constant ZERO_AMOUNT = 0;
+    MockL2ToL2Messenger internal constant MESSENGER = MockL2ToL2Messenger(Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER);
+    IHevm hevm = IHevm(0x7109709ECfa91a80626fF3989D68f67F5b1DD12D);
+
+    address internal remoteToken = address(bytes20(keccak256("remoteToken")));
+    string internal name = "SuperchainERC20";
+    string internal symbol = "SUPER";
+    uint8 internal decimals = 18;
+    address internal user = address(bytes20(keccak256("user")));
+    address internal target = address(bytes20(keccak256("target")));
+
+    OptimismSuperchainERC20 public superchainERC20Impl;
+    OptimismSuperchainERC20 internal optimismSuperchainERC20;
+
+    constructor() {
+        superchainERC20Impl = new OptimismSuperchainERC20();
+        optimismSuperchainERC20 = OptimismSuperchainERC20(
+            address(
+                new ERC1967Proxy(
+                    address(superchainERC20Impl),
+                    abi.encodeCall(OptimismSuperchainERC20.initialize, (remoteToken, name, symbol, decimals))
+                )
+            )
+        );
+
+        // Etch the mocked L2 to L2 Messenger because the `TSTORE` opcode is not supported, and also due to issues with
+        // `encodeVersionedNonce()`
+        address _mockL2ToL2CrossDomainMessenger = address(new MockL2ToL2Messenger(address(optimismSuperchainERC20)));
+        hevm.etch(address(MESSENGER), _mockL2ToL2CrossDomainMessenger.code);
+    }
+
+    function check_setup() public view {
+        assert(optimismSuperchainERC20.remoteToken() == remoteToken);
+        assert(keccak256(abi.encode(optimismSuperchainERC20.name())) == keccak256(abi.encode(name)));
+        assert(keccak256(abi.encode(optimismSuperchainERC20.symbol())) == keccak256(abi.encode(symbol)));
+        assert(optimismSuperchainERC20.decimals() == decimals);
+    }
+
+    /// @custom:property-id 6
+    /// @custom:property-id Calls to sendERC20 succeed as long as caller has enough balance
+    function check_sendERC20SucceedsOnlyIfEnoughBalance(
+        uint256 _balance,
+        uint256 _amount,
+        address _to,
+        uint256 _chainId
+    )
+        public
+    {
+        /* Preconditions */
+        vm.assume(_chainId != CURRENT_CHAIN_ID);
+        vm.assume(_to != address(0));
+
+        // Can't use symbolic value for user since it fails due to `NotConcreteError`
+        // hevm.deal(address(optimismSuperchainERC20), user, _balance);
+        vm.prank(Predeploys.L2_STANDARD_BRIDGE);
+        optimismSuperchainERC20.mint(user, _balance);
+
+        /* Action */
+        vm.prank(user);
+        try optimismSuperchainERC20.sendERC20(_to, _amount, _chainId) {
+            /* Postcondition */
+            assert(_balance >= _amount);
```

Comment:
have you tried mutating this assert? I'm pretty sure the call never succeed as there is no approval made

Reply:
Yes, I tried an `assert(false)` and it indeed fails.
The sender doesn't need to approve since it is a `_burn`, not a transfer

Reply:
my bad, all good